dodiscover/constraint/skeleton.py

```diff
@@ -496,3 +513,119 @@ def _postprocess_ci_test(
             self.sep_set_[y_var][x_var].append(set(cond_set))
             return True
         return False
+
+
+class LearnSemiMarkovianSkeleton(LearnSkeleton):
+    """Learning a semi-markovian skeleton.
+
+    This proceeds by learning a skeleton by testing edges with candidate
```

Comment:
Is there a reference?

What is a "collider oriented DAG"?

Reply:
I suppose we can use the RFCI paper https://arxiv.org/abs/1104.5617 as a reference. However, I think the idea of PDS sets arose earlier. I added some notes and documentation describing this.